core/src/gui/widgets/waterfall.cpp

```diff
@@ -507,7 +506,7 @@ namespace ImGui {
                 doZoom(drawDataStart, drawDataSize, dataWidth, &rawFFTs[((i + currentFFTLine) % waterfallHeight) * rawFFTSize], tempData, _fastFFT);
                 for (int j = 0; j < dataWidth; j++) {
                     pixel = (std::clamp<float>(tempData[j], waterfallMin, waterfallMax) - waterfallMin) / dataRange;
-                    waterfallFb[(i * dataWidth) + j] = waterfallPallet[(int)(pixel * (WATERFALL_RESOLUTION - 1))];
+                    waterfallFb[(i * dataWidth) + j] = waterfallPallet[std::clamp<int>((int)(pixel * (WATERFALL_RESOLUTION - 1)), 0, WATERFALL_RESOLUTION)];
```

Comment:
No need to clamp a second time, the value is already clamped on line 508.

Reply:
There was strange crash, like `pixel` gets somehow wrong value, may be fixed in master, I will check

Reply:
I removed clamp here.